### PR TITLE
Armadillo: new recipe

### PR DESCRIPTION
**Pull request by miqlas**
Provides no tests.

### CODE REVIEW

**Comment by miqlas** [comment]
Something wrong with the hdf5 recipe, as i can't install the generated armadillo package.

My hdf5 package defines himself as:
provides: lib:libhdf5 = 1.8.18 (compatible >= 10)
because it got: libhdf5.so.10.2.1

But the package server says:
nothing provides lib:libhdf5 = 1.8.18.

Maybe i made a mistake in the libhdf5 recipe.

**Comment by pulkomandy** [comment]
There is no relation between the version of the lib and the version of the package. Do not mix them. (sometimes they happend to be identical, but that is not a requirement anywhere). Try this for PROVIDES:

    hdf5 = 1.8.18
    lib:libhdf5 = 10.2.1

and if there is a libhdf5.so.10 in the package:

    lib:libhdf5 = 10.2.1 compat >= 10

**Comment by miqlas** [comment]
@PulkoMandy: thanks for the explanation, i'll fix the hdf5 recipe then.

File: sci-libs/armadillo/armadillo-7.800.2.recipe [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+BUILD()
+{
+	mkdir -p build && cd "$_"
+
+	cmake .. \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DDETECT_HDF5=ON \
+		-DBUILD_SHARED_LIBS=ON \
+		-DINSTALL_INCLUDE_DIR=$includeDir \
+		-DINSTALL_DATA_DIR=$libDir
```
Comment (korli) [inline sci-libs/armadillo/armadillo-7.800.2.recipe]: Why not $dataDir?

File: sci-libs/armadillo/armadillo-7.800.2.recipe [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+INSTALL()
+{
+	cd build
+	make install
+	strip $libDir/lib*
+
+	mkdir -p $libDir/cmake/
+	cp -r $libDir/Armadillo/CMake/. $libDir/cmake/.
```
Comment (korli) [inline sci-libs/armadillo/armadillo-7.800.2.recipe]: Well, is there a recipe which needs armadillo that you could test this against?

**Comment by miqlas** [comment]
DATA_DIR is for .cmake files.
Tested already against mlpack, works ok.

**Pull request merged**